src/garage/_primitives.py

```diff
@@ -0,0 +1,152 @@
+"""Policy interface."""
+import abc
+
+
+class Primitive(abc.ABC):
+    """Primitive interface.
+
+    Args:
+        name (str): Name for logs, variable scopes, etc.
+        spec (garage.InOutSpec): Input/output specification this primitive will
+            satisfy.
+
+    """
+
+    def __init__(self, name, spec):
+        self._name = name
+        self._spec = spec
+
+    @abc.abstractmethod
+    def get_parameters(self):
+        """Get parameters of this primitive.
+
+        Returns:
+            dict[str, np.ndarray]: A dictionary mapping from names to values of
+                each parameter.
+
+        """
+
+    @abc.abstractmethod
+    def set_parameters(self, parameters):
+        """Set the parameters of this primtive.
+
+        Args:
+            parameters (dict[str, np.ndarray]): A dictionary mapping from names
+                to values of each parameter.
+
+        """
+
+    @property
+    def spec(self):
+        """InOutSpec: The input/output spec for this primitive."""
+        return self._spec
+
+    @property
+    def name(self):
+        """str: Name of this primitive."""
+        return self._name
+
+    @abc.abstractmethod
+    def clone(self, name):
+        """Create a fully-independent copy of this primitive.
+
+        Args:
+            name (str): Name for the cloned instance.
+
+        Returns:
+            Primitive: A fully-independent copy of this primitive.
+
+        """
+
+    @abc.abstractmethod
+    def __getstate__(self):
```

Comment:
Does this force this method to be redefined? I agree this method should *work* on this type, but I'm not 100% on how `abc.abstractmethod` works. If another class I inherit from implements this method do I need to re-implement it?

Reply:
yeah, also not sure

Reply:
`abc.abstractmethod` enforces subclass to implements it.

Reply:
i agree this should be removed and instead enforced with some other means, such as using a decorator which checks interfaces